File: testing-modules/testing-libraries/src/main/java/com/baeldung/lambdabehave/Calculator.java
package com.baeldung.lambdabehave;

public class Calculator {

    private int x;
    private int y;

    Calculator(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int add() {
        return this.x + this.y;
    }

    public int divide(int a, int b) {
        return a / b;
    }

    public int add(int a, int b) {
        return a + b;
    }
}

File: core-kotlin-modules/core-kotlin-testing/src/test/kotlin/com/baeldung/junit5/Calculator.kt
package com.baeldung.junit5

class Calculator {
    fun add(a: Int, b: Int) = a + b

    fun divide(a: Int, b: Int) = if (b == 0) {
        throw DivideByZeroException(a)
    } else {
        a / b
    }

    fun square(a: Int) = a * a

    fun squareRoot(a: Int) = Math.sqrt(a.toDouble())

    fun log(base: Int, value: Int) = Math.log(value.toDouble()) / Math.log(base.toDouble())
}
